src/corehost/cli/hostpolicy.cpp

```diff
@@ -116,12 +110,30 @@ int run(const arguments_t& args)
         property_values.push_back(clrjit_path_cstr.data());
     }
 
+    bool set_app_paths = false;
+
+    // Runtime options config properties.
     for (int i = 0; i < g_init.cfg_keys.size(); ++i)
     {
+        // Provide opt-in compatible behavior by using the switch to set APP_PATHS
```

Comment:
I have tested against MusicStore. The app_paths won't be set by default per this fix. The opt-in provided is `$.runtimeOptions.configProperties.Microsoft.NETCore.DotNetHostPolicy.SetAppPaths`

Correct, AFAIK,

1. We shouldn't have an issue here -- simply because `dotnet build` and `dotnet run` cycle doesn't rely on the `APP_PATHS` and fetches everything from NuGet cache.
2. Native assets already are not on `APP_PATHS` for portable apps, so the bug case -- setting this doesn't help.